pkg/k8s/apis/cilium.io/v2/validator/validator.go

```diff
@@ -105,10 +113,91 @@ func (n *NPValidator) ValidateCNP(cnp *unstructured.Unstructured) error {
 	return nil
 }
 
+const (
+	warnEmptyToFromEndpointMessage = "It seems you have a CiliumClusterwideNetworkPolicy " +
+		"with an empty to/from endpoint selector. The behavior of this selector has been " +
+		"changed. The selector now only allows traffic to/from Cilium managed K8s endpoints, " +
+		"instead of acting as a truly empty endpoint selector allowing all traffic. To " +
+		"ensure that the policy behavior does not affect your workloads, consider adding " +
+		"another policy that allows traffic to/from world and cluster entities. For a more " +
+		"detailed discussion on the topic, see GitHub issue #12844"
+)
+
 // ValidateCCNP validates the given CCNP accordingly the CCNP validation schema.
 func (n *NPValidator) ValidateCCNP(ccnp *unstructured.Unstructured) error {
 	if errs := validation.ValidateCustomResource(nil, &ccnp, n.ccnpValidator); len(errs) > 0 {
 		return errs.ToAggregate()
 	}
+
+	logger := n.logger.WithFields(logrus.Fields{
+		logfields.CiliumClusterwideNetworkPolicyName: ccnp.GetName(),
+	})
+
+	// At this point we have validated the custom resource with the new CRV.
+	// We can try converting it to the new CCNP type.
+	// This should not fail so we are ignoring any errors here.
+	ccnpBytes, err := ccnp.MarshalJSON()
+	if err == nil {
+		resCCNP := cilium_v2.CiliumClusterwideNetworkPolicy{}
+		err = json.Unmarshal(ccnpBytes, &resCCNP)
+		if err != nil {
+			logger.Warnf("Error unmarshalling CCNP bytes: %s", err)
+
+			return nil
+		}
+
+		// Print the warninig only once per CCNP.
+		if resCCNP.Spec != nil {
+			if containsEmptyToFromEndpoint(resCCNP.Spec) {
+				logger.Warn(warnEmptyToFromEndpointMessage)
+				return nil
+			}
+		}
+
+		if resCCNP.Specs != nil {
+			for _, rule := range resCCNP.Specs {
+				if containsEmptyToFromEndpoint(rule) {
+					logger.Warn(warnEmptyToFromEndpointMessage)
+					return nil
+				}
+			}
+		}
+	} else {
+		logger.Warnf("Error marshalling CCNP structure: %s", err)
+	}
+
 	return nil
 }
+
+// containsEmptyToFromEndpoint returns true if a CCNP contains an empty endpoint selector
+// in ingress/egress rules.
+// For more information - https://github.com/cilium/cilium/issues/12844#issuecomment-672074170
+func containsEmptyToFromEndpoint(rule *api.Rule) bool {
+	if len(rule.Ingress) > 0 {
+		for _, r := range rule.Ingress {
+			if len(r.FromEndpoints) > 0 {
+				for _, epSel := range r.FromEndpoints {
+					// Print a warning if the endpoint selector is a wildcard.
```

Comment:
This comment is doesn't seem necessary anymore.

Reply:
Oh, I forgot to remove this. :sweat_smile: